File: ToDoApp/toDoFe/src/app/services/todo.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ToDo } from '../model/todo';
import { Observable } from 'rxjs';
import { USER } from '../model/user';

@Injectable({
  providedIn: 'root',
})
export class TodoService {
  //inject HttpClient service in constructor to call backend services
  constructor(private http: HttpClient) {}

  //created addToDo()
  public addToDo(toDo: any): Observable<any> {
    return this.http.post<any>(`http://localhost:9005/toDo/user/addToDo`, toDo);
  }

  //created getAllToDo()
  public getAllToDo(): Observable<any> {
    return this.http.get<any>(`http://localhost:9005/toDo/user/getAllToDo`);
  }

  //created updateToDo()
  public updateToDo(toDo: any): Observable<any> {
    return this.http.put<any>(
      `http://localhost:9005/toDo/user/updateToDo`,
      toDo
    );
  }

  //created deleteToDoByTitle()
  public deleteToDoByTitle(title: any): Observable<any> {
    return this.http.delete<any>(
      `http://localhost:9005/toDo/user/deleteToDoByTitle/${title}`
    );
  }

  //created getToDoByTitle()
  public getToDoByTitle(title: any): Observable<any> {
    return this.http.get<any>(
      `http://localhost:9005/toDo/user/getToDoByTitle/${title}`
    );
  }

  //create getAllListByPriority()
  public getAllListByPriority(priority: any): Observable<any> {
    return this.http.get<any>(
      `http://localhost:9005/toDo/user/getAllListByPriority/${priority}`
    );
  }

  //create getAllToDoByArchiveStatus()
  public getAllToDoByArchiveStatus(isArchive: any): Observable<any> {
    return this.http.get<any>(
      `http://localhost:9005/toDo/user/getAllToDoByArchiveStatus/${isArchive}`
    );
  }

  //create updateTaskArchiveStatus()
  public updateTaskArchiveStatus(toDo: ToDo): Observable<Array<ToDo>> {
    return this.http.put<Array<ToDo>>(
      `http://localhost:9005/toDo/user/updateTaskArchiveStatus`,
      toDo
    );
  }

  //create updateToDoStatus()
  public updateToDoStatus(toDo: any): Observable<any> {
    return this.http.put<any>(
      `http://localhost:9005/toDo/user/updateToDoStatus`,
      toDo
    );
  }

  //create getAllToDoByStatus()
  public getAllToDoByStatus(isCompleted: any): Observable<any> {
    return this.http.get<any>(
      `http://localhost:9005/toDo/user/getAllToDoByStatus/${isCompleted}`
    );
  }

  //create getAllToDoByStatus()
  public getToDoByImportantStatus(isCompleted: any): Observable<any> {
    return this.http.get<any>(
      `http://localhost:9005/toDo/user/getToDoByImportantStatus/${isCompleted}`
    );
  }

  //created deleteTaskByTitle()
  public deleteTaskByTitle(title: any): Observable<any> {
    return this.http.delete<any>(
      `http://localhost:9005/toDo/user/deleteToDoByTitle/${title}`
    );
  }

  //created getAllNotification()
  public getAllNotification(): Observable<any> {
    return this.http.get<any>(
      `http://localhost:9005/notificationController/user/getAllNotification`
    );
  }

  //created deleteAllToDo()
  public deleteAllToDo(): Observable<any> {
    return this.http.delete<any>(
      `http://localhost:9005/toDo/user/deleteAllToDo`
    );
  }

  public getToDOByCompletedStatus(isCompleted: boolean): Observable<any> {
    return this.http.get<any>(
      `http://localhost:9005/toDo/user/getAllToDoByStatus/${isCompleted}`
    );
  }

  //create updateTaskArchiveStatus()
  public updateTaskAsCompletedTask(toDo: any): Observable<any> {
    return this.http.put<any>(
      `http://localhost:9005/toDo/user/updateTaskAsCompletedTask`,
      toDo
    );
  }

  //create updateToDoAsImportantTask()
  public updateToDoAsImportantTask(toDo: any): Observable<any> {
    return this.http.put<any>(
      `http://localhost:9005/toDo/user/updateToDoAsImportantTask`,
      toDo
    );
  }

  //for send register message in gmail
  public sendSimpleEmail() {
    return this.http.get(`http://localhost:9005/user/sendSimpleEmail`);
  }

  //getCurrentUser
  public getCurrentUser(): Observable<USER> {
    return this.http.get<USER>(
      `http://localhost:9005/toDo/user/getCurrentUser`
    );
  }
  public getToDOByDate(): Observable<any> {
    return this.http.get<any>(
      `http://localhost:9005/toDo/user/getToDoByDate`
    );
  }
}

File: ToDoApp/to_do_Backend/ToDoService/src/main/java/com/niit/bej/todoservice/service/ToDoService.java
package com.niit.bej.todoservice.service;

import com.niit.bej.todoservice.domain.ToDo;
import com.niit.bej.todoservice.domain.User;
import com.niit.bej.todoservice.exception.ToDoAlreadyExistException;
import com.niit.bej.todoservice.exception.UserNotFoundException;
import com.niit.bej.todoservice.exception.ToDoNotFoundException;
import com.niit.bej.todoservice.exception.UserAlreadyExistException;

import java.util.List;

public interface ToDoService {
    User registerUser(User user) throws UserAlreadyExistException;
    User addToDo(String emailId, ToDo toDo)throws UserNotFoundException, ToDoAlreadyExistException;

    List<ToDo> getAllToDo(String emailId)throws UserNotFoundException, ToDoNotFoundException;
    User updateToDo(String emailId, ToDo toDo) throws UserNotFoundException,ToDoNotFoundException;
    boolean deleteToDoByTitle(String emailId,String title) throws UserNotFoundException ,ToDoNotFoundException;

    ToDo getToDoByTitle(String emailId, String title) throws UserNotFoundException,ToDoNotFoundException;
    List<ToDo> getAllToDoByPriority(String emailId, String priority) throws UserNotFoundException;
    List<ToDo> getToDoListByArchivedStatus(String emailId,boolean isArchive)throws UserNotFoundException,ToDoNotFoundException;

    User updateTaskAsArchiveStatus(String emailId,ToDo toDo) throws UserNotFoundException,ToDoNotFoundException;

    List<ToDo> getToDoByCompletionStatus(String emailId, Boolean isCompleted) throws ToDoNotFoundException, UserNotFoundException;
    User updateToDoAsCompletedTask(String emailId, ToDo toDo) throws ToDoNotFoundException, UserNotFoundException;

    List<ToDo> getToDoByImportantStatus(String emailId, Boolean isCompleted) throws ToDoNotFoundException, UserNotFoundException;
    User updateToDoAsImportantTask(String emailId, ToDo toDo) throws ToDoNotFoundException, UserNotFoundException;
    boolean deleteAllToDo(String emailId)throws UserNotFoundException,ToDoNotFoundException;
    User getCurrentUser(String emailId) throws UserNotFoundException;
    List<ToDo> getAllToDoByDate(String emailId)throws UserNotFoundException, ToDoNotFoundException;
}
